decidim-core/lib/decidim/coauthorable.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module Decidim
+  # This concern contains the logic related to collective or shared authorship.
+  #
+  # Coauthorable shares nearly the same object interface as Authorable but with some differences.
+  # - `authored_by?(user)` is exactly the same
+  # - `normalized_authors` is pluralized
+  #
+  # All coauthors, including the initial author, will share the same permissions.
+  module Coauthorable
+    extend ActiveSupport::Concern
+
+    included do
+      has_many :coauthorships, as: :coauthorable, class_name: "Decidim::Coauthorship", dependent: :destroy
+      has_many :authors, through: :coauthorships, class_name: "Decidim::User"
+      has_many :user_groups, as: :coauthorable, class_name: "Decidim::UserGroup", through: :coauthorships
+
+      scope :from_author, ->(author) { joins(:coauthorships).where("decidim_coauthorships.decidim_author_id": author.id) }
+      scope :from_user_group, ->(user_group) { joins(:coauthorships).where("decidim_coauthorships.decidim_user_group_id": user_group.id) }
+
+      # Checks whether the user is author of the given proposal, either directly
+      # authoring it or via a user group.
+      #
+      # user - the user to check for authorship
+      def authored_by?(user)
+        coauthorships.where(author: user).or(coauthorships.where(user_group: user.user_groups)).exists?
```

Comment:
Same about the `decidim_user_group_id` column in the first part of the query

Reply:
If I leave the first part as it is now, I even guess if the second part of the query is even required, let me check!